src/main/java/com/xmslyz/luhn/logic/LuhnChecksumComputing.java

```diff
@@ -8,18 +8,17 @@ public class LuhnChecksumComputing implements Computation {
 
     @Override
     public String compute(String input) {
-        try {
-            this.number = input.trim().replace(" ", "");
-            setArray(number.length());
-            multiplyWages();
-            countTotal();
-            getChecksum();
-            return this.checkSum;
-        } catch (NumberFormatException | NullPointerException e) {
-            System.out.println("Input error.");
+        if (input == null) throw new NullPointerException("Expected not-null input");
+        if (input.isEmpty()) throw new IllegalArgumentException("Expected empty string");
+        this.number = input.trim().replace(" ", "");
+        if (Long.parseLong(number) < 0 ) throw new NumberFormatException("Expected number < 0");
```

Comment:
Używając `Long` sztucznie zaniżasz wielkość liczby, którą Twój algorytm może przeprocesować.  
Klasa `BigInteger` może przechowywać liczbę całkowitą dowolnej długości (o ile pamięci w komputerze starczy) - i to jej powinieneś tutaj użyć:
```suggestion
        if (new BigInteger(number).signum() < 0) throw new NumberFormatException("Expected number >= 0");
```
`new BigInteger(number)` stworzy instancję klasy BigInteger i jeśli tylko string `number` zawiera nieodpowiednie znaki, rzucony zostanie `NumberFormatException`. 
Następnie na nowostworzonej instancji BigInteger wołana jest metoda `signum`, której dokumentacja mówi, że:
```
returns -1, 0 or 1 as the value of this BigInteger is negative, zero or positive
```
więc wywołanie `.signum() < 0` sprawdza czy podana liczba jest negatywna.
Żeby używać BigInteger musisz go zaimportować: import java.math.BigInteger;`